byway/src/main/java/com/google/sps/servlets/EntityServlet.java

```diff
@@ -0,0 +1,115 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Trip;
+import com.google.sps.data.UserInfo;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/entity")
+public class EntityServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    com.google.appengine.api.users.User user = userService.getCurrentUser();
+    UserInfo userInfo = addUserEntity(user);
+    List<String> userTripIds = userInfo.getTripIds();
+    ArrayList<Trip> userTrips = new ArrayList<Trip>();
+    for (String tripId : userTripIds) {
+      try {
+        Entity tripEntity = datastore.get(KeyFactory.stringToKey(tripId));
+        Trip trip = Trip.fromEntity(tripEntity);
+        userTrips.add(trip);
+      } catch (EntityNotFoundException exception) {
+        return;
```

Comment:
Note that at this point, you haven't set any error codes, haven't written any data to the result, and haven't logged the error, so if you just return here, then any time any single trip is missing from the datastore, you'll just get back an empty result with no way of knowing what trip was missing. You should probably 1) set an error code on the response (404 for not found might be appropriate, or if your server is supposed to guarantee that all trips on the UserInfo object always exist as entities in the datastore, then it might be preferable to return 500 for server-error, since a missing trip would be a mistake somewhere in the server). 2) log an error here so you know *which* trip was missing and what the error message was.

Reply:
done, added a 500 error and a flogger message